### PR TITLE
version 2.4.0 to main

### PR DESCRIPTION
**Pull request by paolocappelletti**

### CODE REVIEW

File: ci/publish.sh [hunk 1]
```diff
@@ -1,16 +1,21 @@
 #!/bin/bash
 
 set -eo pipefail
+set -x
```
Comment (lander86) [inline ci/publish.sh]: You should unser the set -x ; probably it was here for testing purposes

File: ci/setup_env.sh [hunk 2]
```diff
@@ -1,9 +1,13 @@
 #!/bin/bash
 
 set -eo pipefail
+set -x
```
Comment (lander86) [inline ci/setup_env.sh]: same

**Review by lander86 — COMMENTED** [review]
Please remove the set -x

**Review by saratnt — APPROVED** [review]

**Review by otoumas — APPROVED** [review]
LGTM

**Pull request merged**